ext/fetch/20_headers.js

```diff
@@ -91,6 +92,32 @@ function fillHeaders(headers, object) {
 // deno-lint-ignore no-control-regex
 const ILLEGAL_VALUE_CHARS = new SafeRegExp(/[\x00\x0A\x0D]/);
 
+function checkForInvalidValueChars(value) {
+  for (let i = 0; i < value.length; i++) {
+    const c = StringPrototypeCharCodeAt(value, i);
+
+    if (c === 0x0a || c === 0x0d || c === 0x00) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+// TODO(bartlomieju): make it LRU cache with 64/128 size limit.
+const cache = {};
+
+function checkHeaderNameForHttpTokenCodePoint(name) {
+  const cachedResult = cache[name];
+  if (typeof cachedResult !== "undefined") {
+    return cachedResult;
+  }
+
+  const valid = RegExpPrototypeExec(HTTP_TOKEN_CODE_POINT_RE, name) !== null;
```

Comment:
I think we should add a very simple clear mechanic - `if (map.size > 4096) map.clear();` for example

Reply:
Okay, applied this suggestion

Reply:
hello, I think the 4096 in this block should not be a magic number, but should be named a constant to indicate its specific meaning @bartlomieju @lucacasonato

Reply:
@markthree sorry, I missed your comment before landing this PR. Feel free to open a PR that changes this bit.

Reply:
> @markthree sorry, I missed your comment before landing this PR. Feel free to open a PR that changes this bit. 对不起，我错过了你的评论之前登陆这个公关。请随意打开一个PR，改变这一点。

Okay, I have mentioned pr.